src/python/tests/storage/test_storage_api.py

```diff
@@ -0,0 +1,283 @@
+        assert result["rows"][0]["description"] == "Testing the storage service data 1"
+        assert result["rows"][0]["value"] == 10
+        assert result["rows"][0]["previous_value"] == 2
+
+    def test_limit(self):
+        result = Storage().query_tbl_with_payload("statistics", PayloadBuilder().LIMIT(1).payload())
+        assert len(result["rows"]) == 1
+        assert result["count"] == 1
+        assert result["rows"][0]["key"] == "TEST_1"
+        assert result["rows"][0]["description"] == "Testing the storage service data 1"
+        assert result["rows"][0]["value"] == 10
+        assert result["rows"][0]["previous_value"] == 2
+
+    @pytest.mark.skip(reason="FOGL-607 #4")
+    def test_offset(self):
+        pass
+
+    def test_limit_offset(self):
+        result = Storage().query_tbl_with_payload("statistics", PayloadBuilder().LIMIT(2).OFFSET(1).payload())
+        assert len(result["rows"]) == 1
+        assert result["count"] == 1
+        assert result["rows"][0]["key"] == "TEST_2"
+        assert result["rows"][0]["description"] == "Testing the storage service data 2"
+        assert result["rows"][0]["value"] == 15
+        assert result["rows"][0]["previous_value"] == 2
+
+    def test_order(self):
+        result = Storage().query_tbl_with_payload("statistics", PayloadBuilder().
+                                                  ORDER_BY(["key", "desc"]).payload())
+        assert len(result["rows"]) == 2
+        assert result["count"] == 2
+        assert result["rows"][0]["key"] == "TEST_2"
+        assert result["rows"][0]["description"] == "Testing the storage service data 2"
+        assert result["rows"][0]["value"] == 15
+        assert result["rows"][0]["previous_value"] == 2
+
+    def test_aggregate(self):
+        result = Storage().query_tbl_with_payload("statistics", PayloadBuilder().
+                                                  AGGREGATE(["max", "value"]).payload())
+        assert len(result["rows"]) == 1
+        assert result["count"] == 1
+        assert result["rows"][0]["max_value"] == 15
+
+    @pytest.mark.skip(reason="FOGL-617")
+    def test_group(self):
+        result = Storage().query_tbl_with_payload("statistics", PayloadBuilder().SELECT("previous_value").
+                                                  GROUP_BY("previous_value").payload())
+        assert len(result["rows"]) == 1
+        assert result["count"] == 1
+        assert result["rows"][0]["previous_value"] == 2
+
+    def test_aggregate_group(self):
+        result = Storage().query_tbl_with_payload("statistics", PayloadBuilder().
+                                                  AGGREGATE(["min", "previous_value"]).
+                                                  GROUP_BY("previous_value").payload())
+        assert len(result["rows"]) == 1
+        assert result["count"] == 1
+        assert result["rows"][0]["min_previous_value"] == 2
+        assert result["rows"][0]["previous_value"] == 2
+
```

Comment:
test group_by with having; as having woks with and without

Reply:
done